src/V3ThreadPool.h

```diff
@@ -109,22 +109,44 @@ class V3ThreadPool final {
     // Indicates whether multithreading has been suspended.
     // Used for error detection in resumeMultithreading only. You probably should use
     // m_exclusiveAccess for information whether something should be run in current thread.
-    bool m_multithreadingSuspended VL_GUARDED_BY(m_mutex) = false;
+    std::atomic_bool m_multithreadingSuspended{false};
 
     // CONSTRUCTORS
     V3ThreadPool() = default;
     ~V3ThreadPool() {
+        if (m_multithreadingSuspended) {
+            // Ideally we shouldn't deal with this and just call the std::abort. However,
+            // std::exit(0) (which invokes this destructor) is called in multiple places with
+            // multithreading being suspended. Aborting would change the exit code, what would not
+            // be acceptable.
+            resumeMultithreading();
+        }
+        // Unexpected states. Aborting to subtly signal to the outside world that something is
+        // wrong. This won't make things worse to an user - the program is already terminating at
+        // this point anyway, most likely as a result of an error. Using if/abort instead of assert
+        // because assert can be disabled.
+        if (m_exclusiveAccess) std::abort();
+        if (m_stopRequested) std::abort();
```

Comment:
Use VL_UNCOVERABLE(condition).  Print some message to help the user before aborting.  Similar below please.

Should we instead be using V3ErrorGuarded::vlAbortOrExit()?

Reply:
> Should we instead be using V3ErrorGuarded::vlAbortOrExit()?

We could if we keep the `vlAbortOrExit()` implementation from this PR. However, if we replace `quick_exit(1)` with anything that executes destructors of static objects, I think we'll end with endless recursion or at least undefined behavior. `V3Error::vlAbort()` should be fine in any case.

> Print some message to help the user before aborting.

The abort (direct or via `vlAbort`) has the benefit of returning with dedicated exit code instead of `1` as `vlAbortOrExit()` does, which leads to a message about aborting being printed by `bin/verilator`. Is this message enough, or should we print additional message in the destructor?